Emux.GameBoy/GameBoy.cs

```diff
@@ -14,10 +16,50 @@ namespace Emux.GameBoy
     /// </summary>
     public class GameBoy
     {
-        public GameBoy(ICartridge cartridge, IClock clock, bool preferGbcMode)
+		public const float OfficialFrameRate = 59.7f;
+
+		private readonly ManualResetEvent _continueSignal = new ManualResetEvent(false);
+		private readonly ManualResetEvent _terminateSignal = new ManualResetEvent(false);
+		private readonly ManualResetEvent _frameStartSignal = new ManualResetEvent(false);
+		private readonly ManualResetEvent _breakSignal = new ManualResetEvent(false);
+
+		/// <summary>
+		/// Occurs when the processor has resumed execution.
+		/// </summary>
+		public event EventHandler Resumed;
+
+		/// <summary>
+		/// Occurs when the processor is paused by breaking the execution explicitly, or when the control flow hit a breakpoint.
+		/// </summary>
+		public event EventHandler Paused;
+
+		/// <summary>
+		/// Occurs when the process has completely shut down.
+		/// </summary>
+		public event EventHandler Terminated;
+
+		public event StepEventHandler PerformedStep;
+
+		private readonly IDictionary<ushort, Breakpoint> _breakpoints = new Dictionary<ushort, Breakpoint>();
+
+		private IClock Clock;
+
+		private int framesCount;
+
+		public double FramesPerSecond;
+		public bool EnableFrameLimit = true;
+		public TimeSpan FrameDelta;
```

Comment:
These should either be public properties or private fields using the same naming convention as everywhere else.